### PR TITLE
feat: separated file uploader

### PR DESCRIPTION
**Pull request by CzarekDryl**
https://tw.auroracreation.com/app/tasks/15580149

image

image

### CODE REVIEW

File: src/components/v5/common/AvatarUploader/AvatarUploader.tsx [hunk 1]
```diff
@@ -39,31 +42,57 @@ const AvatarUploader: FC<AvatarUploaderProps> = ({
...
-        handleFileAccept={handleFileUpload}
-        handleFileReject={handleFileReject}
-        handleFileRemove={handleFileRemove}
-        placeholder={getPlaceholder(isLoading, avatarPlaceholder)}
-        forwardedRef={dropzoneRef}
-        errorCode={uploadAvatarError}
-        isAvatarUploaded={user?.profile?.avatar !== null}
-        isPropgressContentVisible={showPropgress}
-      />
+  const { fileDimension, fileFormat, fileSize } = fileOptions;
```
Comment (krawczykkatarzyna) [inline src/components/v5/common/AvatarUploader/AvatarUploader.tsx:45]: this functionality is repeated here and in ChangeColonyLogo.tsx

File: src/components/v5/common/ActionsContent/partials/ColonyDetailsFields/partials/ChangeColonyLogo.tsx [hunk 2]
```diff
@@ -65,33 +90,36 @@ const ChangeColonyLogo: FC = () => {
...
             handleFileRemove={handleFileRemove}
+            fileOptions={fileOptions}
+            forwardedRef={dropzoneRef}
+            errorCode={avatarFileError}
+            isAvatarUploaded={colonyAvatarImage !== null}
+            isPropgressContentVisible={showPropgress}
           />
-        )}
-      </>
+
```
Comment (joanna-pagepro) [inline src/components/v5/common/ActionsContent/partials/ColonyDetailsFields/partials/ChangeColonyLogo.tsx:112]: Please remove redundant empty lines

File: src/components/v5/common/ActionsContent/partials/ColonyDetailsFields/partials/ChangeColonyLogo.tsx [hunk 3]
```diff
@@ -65,33 +90,36 @@ const ChangeColonyLogo: FC = () => {
...
-          isPropgressContentVisible={showPropgress}
-        />
-
-        {showPropgress && (
-          <ProgressContent
-            progress={uploadProgress}
-            fileName={file.fileName}
-            fileSize={file.fileSize}
+      <div className="flex sm:flex-row flex-col gap-4 sm:gap-2">
+        {avatar}
```
Comment (joanna-pagepro) [inline src/components/v5/common/ActionsContent/partials/ColonyDetailsFields/partials/ChangeColonyLogo.tsx:94]: Is this used in any other place?

File: src/components/v5/common/ActionsContent/partials/ColonyDetailsFields/partials/ChangeColonyLogo.tsx [hunk 4]
```diff
@@ -65,33 +90,36 @@ const ChangeColonyLogo: FC = () => {
...
+        <div className="flex flex-col gap-2 w-full">
+          <div className="hidden sm:block text-gray-600 text-sm">
```
Comment (joanna-pagepro) [inline src/components/v5/common/ActionsContent/partials/ColonyDetailsFields/partials/ChangeColonyLogo.tsx:96]: Why is it a div?

File: src/components/v5/common/AvatarUploader/partials/AvatarUploader.module.css [hunk 5]
```diff
@@ -1,7 +1,3 @@
-.text {
-  @apply text-gray-600 text-sm;
-}
-
 .wrapper {
```
Comment (joanna-pagepro) [inline src/components/v5/common/AvatarUploader/partials/AvatarUploader.module.css:1]: Can you also get rid of other classes?

File: src/components/v5/common/AvatarUploader/partials/DefaultContent.tsx [hunk 6]
```diff
@@ -6,52 +6,85 @@ import Icon from '~shared/Icon';
...
+          </span>
         </button>
-      </div>
-      <span className={styles.text}>
-        {formatMessage({ id: 'avatar.uploader.text' })}
-      </span>
+      ) : (
+        <>
+          <div className={styles.iconWrapper}>
+            <div className={`bg-gray-50 w-9 h-9 ${styles.iconCircle}`}>
```
Comment (joanna-pagepro) [inline src/components/v5/common/AvatarUploader/partials/DefaultContent.tsx:48]: You can use clsx in such cases, but please try to move the styles from the file, directly to the item

File: src/components/v5/common/AvatarUploader/types.ts [hunk 7]
```diff
@@ -23,10 +23,15 @@ export type ErrorContentProps = Pick<
   getInputProps: <T extends DropzoneInputProps>(props?: T | undefined) => T;
 };
 
+export interface FileUploadOptions {
+  fileFormat: string[];
+  fileSize: string;
+  fileDimension: string;
+}
+
 export type FileUploadProps = {
```
Comment (joanna-pagepro) [inline src/components/v5/common/AvatarUploader/types.ts:32]: Please change all types to interfaces

File: src/i18n/en.json [hunk 8]
```diff
@@ -970,8 +970,8 @@
   "error.usernameTaken": "Username not available",
   "error.validEmail": "Provide a valid email address",
   "success.userName": "Username available",
-  "avatar.uploader.info": "Must be .PNG, .JPG or .SVG, at least 250x250px in size and can not exceed 1MB.",
-  "avatar.uploader.text": ".PNG, .JPG or .SVG, 250x250px and max file size 1MB",
+  "avatar.uploader.info": "Must be {format}, at least {dimension} in size and can not exceed {size}.",
```
Comment (joanna-pagepro) [inline src/i18n/en.json:973]: 👍

File: src/stories/common/FileUpload.stories.tsx [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+    layout: 'padded',
+  },
+  decorators: [
+    (Story) => (
+      <div className="max-w-[36.5rem]">
+        <Story />
+      </div>
+    ),
+  ],
+  argTypes: {},
```
Comment (joanna-pagepro) [inline src/stories/common/FileUpload.stories.tsx:19]: Please remove this field if it's not used

**Review by joanna-pagepro — CHANGES_REQUESTED** [review]

**Review by krawczykkatarzyna — APPROVED** [review]

**Review by joanna-pagepro — APPROVED** [review]

**Pull request merged**